skills/slashCommands.js

```diff
@@ -62,39 +62,42 @@ const getMessageFromRes = (res, learn) => {
     (a, b) => b - a
   );
 
-  const firstPlace = res.filter(row => row.count === uniqueSortedPoints[0]);
-  const secondPlace =
-    uniqueSortedPoints.length >= 2 &&
-    res.filter(row => row.count === uniqueSortedPoints[1]);
-  const thirdPlace =
-    uniqueSortedPoints.length >= 3 &&
-    res.filter(row => row.count === uniqueSortedPoints[2]);
-
-  let message = `*:tada: Congratulations :tada: to our best ${
-    learn ? 'learners' : 'teachers'
-  } this week*:
-
-  *${firstPlace.map(obj => obj.username).join(', ')}* ${
-  learn ? 'learned' : 'taught'
-} *${uniqueSortedPoints[0]}* ${
-  uniqueSortedPoints[0] === '1' ? 'thing' : 'things'
-}`;
-
-  if (secondPlace) {
-    message += `, *${secondPlace.map(obj => obj.username).join(', ')}* ${
-      learn ? 'learned' : 'taught'
-    } *${uniqueSortedPoints[1]}* ${
-      uniqueSortedPoints[1] === '1' ? 'thing' : 'things'
-    }`;
+  const role = learn ? 'learners' : 'teachers';
+  let message = `*:tada: Congratulations :tada: to our best ${role} this week*:\n\n`;
+
+  const scores = getScores(res, uniqueSortedPoints);
+  scores.forEach(score => {
+    message += getIndividualMessage(score, learn);
+  });
+
+  return message;
+};
+
+const getScores = (res, uniqueSortedPoints) => {
+  let scores = [];
+  const numScores = Math.min(3, uniqueSortedPoints.length);
+  for (let i = 0; i < numScores; i++) {
+    const numPoints = uniqueSortedPoints[i];
+    const people = res.filter(row => row.count === numPoints);
+    const usernames = people.map(obj => obj.username).join(', ');
+    scores.push({ position: i + 1, usernames, numPoints });
   }
```

Comment:
Same here, but at least the `let` is buried inside a function so it's _way_ harder to pollute the state.
It's also harder to turn this to a reduce :)

Reply:
I had a crack at reducing it anyway, seems to work.